### DISCUSSION TITLE
Use of orkes queues does that change license from Apache 2.0?

### DISCUSSION THREAD

**Discussion by Anupchat** [conductor-oss/conductor#108] [General]
I see by default we have orkes queues enabled - https://github.com/conductor-oss/conductor/blob/495eac67b3f59b743e7bf4f0e281f7764502a848/server/build.gradle#L71
and 
https://github.com/conductor-oss/conductor/blob/495eac67b3f59b743e7bf4f0e281f7764502a848/server/src/main/java/com/netflix/conductor/Conductor.java#L31

The License for Orkes is not Apache 2.0 https://github.com/orkes-io/orkes-queues/blob/main/LICENSE.txt 

Is there any issue with this setup? 
If the default queue is going to be Orkes then can we also get the license matched?

**Comment by Prebiusta** [comment]
[Orkes license](https://github.com/orkes-io/orkes-queues/blob/9665771cbd4340aaf046b64c6e834efe3140d8d9/LICENSE.txt#L101-L106) mentions the following

> For clarity, the foregoing does not prohibit Licensee from combining
         and using the Software with Conductor code that is subject only to the Apache 2.0
         license or subject to an Orkes community license.

As I understand, queues can be used as part of Conductor code

**Reply by Anupchat** [reply]
Thanks for the clarification.

**Comment by Anupchat** [comment]
Can someone from Orkes also confirm this? @v1r3n?

**Comment by Anupchat** [comment]
@v1r3n can you answer. Legal is pointing out problems with this setup if part of the license is Apache 2 for conductor and orkes-queues isn't especially since the library needs to be included into the server. Can we get orkes-queues license to match Apache 2.0? That would also help people taking up replacement for dyno-queues which was also Apache 2 license.

**Comment by Anupchat** [comment]
This is the response from legal -

> The ambiguity comes from section 2.1: Software License. As of the Effective Date, subject to the terms and conditions of this Agreement, including, without limitation, Article 3 (License Restrictions and Intellectual Property Rights), Orkes grants Licensee a limited, non-exclusive, non-transferable, and non-sublicensable license, to:
>           (a)    evaluate the Software in Licensee’s development and testing environments;
>           (b)    use the Software internally for Licensee’s internal business purposes;
>           (c)    modify and make derivative works of the Software;
>           (d)    reproduce the Software; and
>           (e)    distribute Software in binary or Source Code form.
> 
> Although they provide distribution rights (e), there seems to be limitations on the use of the software for evaluation and for internal business purposes (a and b).  
>
